rl4co/envs/scheduling/fjsp/env.py

```diff
@@ -0,0 +1,436 @@
+
+    def _make_spec(self, generator: FJSPGenerator):
+        self.observation_spec = CompositeSpec(
+            time=UnboundedDiscreteTensorSpec(
+                shape=(1,),
+                dtype=torch.int64,
+            ),
+            next_op=UnboundedDiscreteTensorSpec(
+                shape=(self.num_jobs,),
+                dtype=torch.int64,
+            ),
+            proc_times=UnboundedDiscreteTensorSpec(
+                shape=(self.num_mas, self.n_ops_max),
+                dtype=torch.float32,
+            ),
+            pad_mask=UnboundedDiscreteTensorSpec(
+                shape=(self.num_mas, self.n_ops_max),
+                dtype=torch.bool,
+            ),
+            start_op_per_job=UnboundedDiscreteTensorSpec(
+                shape=(self.num_jobs,),
+                dtype=torch.bool,
+            ),
+            end_op_per_job=UnboundedDiscreteTensorSpec(
+                shape=(self.num_jobs,),
+                dtype=torch.bool,
+            ),
+            start_times=UnboundedDiscreteTensorSpec(
+                shape=(self.n_ops_max,),
+                dtype=torch.int64,
+            ),
+            finish_times=UnboundedDiscreteTensorSpec(
+                shape=(self.n_ops_max,),
+                dtype=torch.int64,
+            ),
+            job_ops_adj=UnboundedDiscreteTensorSpec(
+                shape=(self.num_jobs, self.n_ops_max),
+                dtype=torch.int64,
+            ),
+            ops_job_map=UnboundedDiscreteTensorSpec(
+                shape=(self.n_ops_max),
+                dtype=torch.int64,
+            ),
+            ops_sequence_order=UnboundedDiscreteTensorSpec(
+                shape=(self.n_ops_max),
+                dtype=torch.int64,
+            ),
+            ma_assignment=UnboundedDiscreteTensorSpec(
+                shape=(self.num_mas, self.n_ops_max),
+                dtype=torch.int64,
+            ),
+            busy_until=UnboundedDiscreteTensorSpec(
+                shape=(self.num_mas,),
+                dtype=torch.int64,
+            ),
+            num_eligible=UnboundedDiscreteTensorSpec(
+                shape=(self.n_ops_max,),
+                dtype=torch.int64,
+            ),
+            job_in_process=UnboundedDiscreteTensorSpec(
+                shape=(self.num_jobs,),
+                dtype=torch.bool,
+            ),
+            job_done=UnboundedDiscreteTensorSpec(
+                shape=(self.num_jobs,),
+                dtype=torch.bool,
+            ),
+            shape=(),
+        )
+        self.action_spec = BoundedTensorSpec(
+            shape=(1,),
+            dtype=torch.int64,
+            low=-1,
+            high=self.n_ops_max,
+        )
+        self.reward_spec = UnboundedContinuousTensorSpec(shape=(1,))
+        self.done_spec = UnboundedDiscreteTensorSpec(shape=(1,), dtype=torch.bool)
+
+    @staticmethod
+    def render(td, idx):
+        return render(td, idx)
```

Comment:
I played with this render but I didn't get an output 🤔. Will this render plot actions? But this is in low priority, don't worry.

Reply:
hm that's weird, for me it works (in vscode). It should plot the (partial) schedule, like this for example:
![image](https://github.com/ai4co/rl4co/assets/66203600/b252e10a-7290-40d9-8f57-bbacc6a81eb7)